### PR TITLE
Update matplotlib canonical link

### PR DESCRIPTION
**Pull request by jklymak**
Please see: https://discourse.matplotlib.org/t/canonical-documentation-have-moved/21863

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Hi @jklymak , thanks for fixing this. I note that we have 20 other occurrences of the url ` https://matplotlib.org/` in the repo (` git grep https://matplotlib.org/)`, some are pointing to specific versions of matplotlib but the majority isn't. Should those be changed as well ?

**Comment by jklymak** [comment]
Up to you - the old links should work via html refresh redirect - but the intersphinx link checker doesn't like the redirects and complains, so I was proactively pinging a few projects.  Thanks!

**Comment by neutrinoceros** [comment]
Wrote a cute one-line to perform this sweeping change, can you run it ? or otherwise allow me to push to your branch ?

```bash
$ git ls-files --recurse-submodules | xargs egrep "https://matplotlib.org/[^3]" | cut -d ':' -f 1 | uniq | xargs sed -i -e "s|https://matplotlib.org|https://matplotlib.org/stable|g"
```

**Comment by jklymak** [comment]
Feel free to push to branch or close this and start fresh.  I just made this change directly on GitHub and don't have yt downloaded locally (I'd lose my oceanographer license 😉 )

**Comment by matthewturk** [comment]
@jklymak we really really want to talk a lot more about oceanography!  I really hope the two are not in conflict.  :)

**Comment by neutrinoceros** [comment]
pushed :)

File: doc/source/cookbook/complex_plots.rst [hunk 1]
```diff
@@ -88,7 +88,7 @@ Multipanel with Axes Labels
 This illustrates how to use a SlicePlot to control a multipanel plot.  This
 plot uses axes labels to illustrate the length scales in the plot.
 See :ref:`slice-plots` and the
-`Matplotlib AxesGrid Object <https://matplotlib.org/mpl_toolkits/axes_grid/api/axes_grid_api.html>`_
+`Matplotlib AxesGrid Object <https://matplotlib.org/stable/mpl_toolkits/axes_grid/api/axes_grid_api.html>`_
```
Comment (cphyc) [inline doc/source/cookbook/complex_plots.rst:91]: Oh no 404!

Comment (neutrinoceros) [reply doc/source/cookbook/complex_plots.rst:91]: goof catch. I figured my one-liner could use more refinement but I'm now just fixing the 404 ones manually

**Comment by neutrinoceros** [comment]
I think I reverted all the 404 errors, thanks @cphyc

**Comment by Xarthisius** [comment]
@yt-fido test this please

**Review by cphyc — APPROVED** [review]

**Comment by cphyc** [comment]
I think I tested all URLs and they're all non-404!

**Pull request merged**